jni-glue/src/string_chars.rs

```diff
@@ -27,16 +27,35 @@ impl<'env> StringChars<'env> {
         }
     }
 
+    /// Construct a StringChars from an Env + AsRef<str>
+    pub fn from_env_str<S: AsRef<str>>(env: &'env Env, string: S) -> Self {
+        let chars = string.as_ref().encode_utf16().collect::<Vec<_>>();
+        let string = unsafe { env.new_string(
+            chars.as_ptr() as *const jchar,
+            chars.len() as jsize,
+        ) };
+        unsafe { Self::from_env_jstring(env, string) }
+    }
```

Comment:
I think I'd much rather move this logic directly into java::lang::String::from_env_str.  StringChars is really only means as an iterator-esque jstring reader along the lines of [std::str::Chars](https://doc.rust-lang.org/std/str/struct.Chars.html), not a general purpouse reader/writer interop type.

This would also get rid of the desire/need to make `chars` a cell / adding extra conditional logic - it might be nice to make this `Send`+`Sync` in the future.

Reply:
I completely agree.

But I don't sure we really need StringChars as a proxy because rust has a deal with utf8 strings, not utf16.